test/server/http_filter_integration_test_base.h

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <string>
+
+#include "external/envoy/test/integration/http_integration.h"
+
+namespace Nighthawk {
+
+/**
+ * Base class with shared functionality for testing Nighthawk test server http filter extensions.
+ */
+class HttpFilterIntegrationTestBase : public Envoy::HttpIntegrationTest {
+protected:
+  /**
+   * Indicate the expected response origin.
+   */
+  enum class ResponseOrigin { UPSTREAM, EXTENSION };
```

Comment:
Can we add a one line summary here for each type of ResponseOrigin?  This may also allow you to remove a comment elsewhere, if it's duplication at this point.